### DISCUSSION TITLE
1MB buffer on the stack?

### DISCUSSION THREAD

**Discussion by JahButNah** [Disservin/chess-library#87] [Ideas]
I know you want speed, but you can't have my stack. In Visual C++ the default Stack Reserve Size is 1MB in virtual memory and the Stack Commit Size is 4KB in physical memory.
I had to hunt down the excessive stack request by climbing into your implementation.
StreamParser<N>::StreamBuffer creates an NxN std::array with a default of N=1KiB.
Also, maybe StreamParser<N>::StreamBuffer::fill() should use stream_.read(..., buffer_.size()) instead of duplicating the "N * N" from the std::array<...> definition.

Thanks for your attention.

**Comment by Disservin** [comment]
The `StreamParser` is templated and can be configured with the buffer size, so just define a lower value?

**Reply by Disservin** [reply]
the default could be lower though..

**Comment by JahButNah** [comment]
True. The squaring could be done without, though. It's only confusing.
